### ISSUE TITLE
loading "libpcsclite_real.so.1" failed: Seems to affect sudo and polkit in GNome

### ISSUE THREAD

**Issue by Lyndeno** [NixOS/nixpkgs#309158]
### Describe the bug
When running pkexec within Gnome:
```console
$ journalctl
May 04 18:55:28 neo .gnome-shell-wr[2649]: Unknown line 'loading "libpcsclite_real.so.1" failed: libpcsclite_real.so.>
```

Gnome authentication window never opens, meaning I cannot authenticate. Same as if I use polkit for 1Password.

Similar message is printed in console when running sudo

### Steps To Reproduce
1. Attempt action that requires polkit authentication

### Screenshots
Gnome authentication window opens.

No errors when using sudo
### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.30, NixOS, 24.05 (Uakari), 24.05.20240504.9a22fec`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/a1bhfahqhc6cbj5hc7wq00rikm7vy4d2-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by MarkRijckenberg** [comment]
This issue with libpcsclite_real.so.1 is also causing the build of eid-mw version 5.1.16 to fail:

https://hydra.nixos.org/build/258123099/nixlog/1/tail

**Comment by yshui** [comment]
This also breaks gpg's smartcard support.

**Comment by NovaViper** [comment]
@yshui Can confirm this aswell. Tried to update my flake earlier and noticed my Yubikey no longer functioned with gpg with this in the logs when I run `gpg --card-status`
```
May 07 22:33:39 yoganova gpg-agent[4913]: scdaemon[4913]: pcsc_establish_context failed: internal error (0x80100001)
```

Affected version info
```
❯ nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.9, NixOS, 24.05 (Uakari), 24.05.20240507.951e023`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/cx3n6rqiyyf1zjclyz341h09dbb4s5f4-source`
```

**Comment by jessesung** [comment]
Got the same error when trying to use `systemd-cryptenroll` to add a FIDO2 key.

**Comment by arjan-s** [comment]
Getting the same error when using the `pcsc-safenet` PKCS#11 provider in OpenVPN:
```
mei 13 08:18:51 laptop openvpn[4428]: loading "libpcsclite_real.so.1" failed: libpcsclite_real.so.1: cannot open shared object file: No such file or directory
```

**Comment by Lyndeno** [comment]
Issue fixed in #308884 and thus #309297 for me
